drt_behavior demo (#37)
NASA-JSC-Robotics/clr_ws: src/clr_sim_demos/clr_behavior_pick_and_place_demo/README.md

# CLR BehaviorTree Pick and Place Demo

> Similar to the `clr_pick_and_place_demo` package, this package contains an example of the CLR robot picking up and placing a cargo transfer bag from it's spawned location to the top of the bench.

Reviewer: Let's assume one will eventually be retired...

```suggestion
This package contains an example of the CLR robot picking up and placing a cargo transfer bag from its spawned location to the top of the bench.
```

> The purpose of this package is demonstration capability using behavior trees and the `drt_behaviors` package.

Reviewer: ```suggestion
The purpose of this package is demonstrations of capabilities using behavior trees and the `drt_behaviors` package.
```

For an introduction to behavior tree and their utilization, as well as the core packages that we build off, please refer to https://www.behaviortree.dev/.

## Instructions

### Setting up

> 1. Start up the mujoco simulation, the default flag will include the mockup environment with the robot.

Reviewer: FWIW I recommend the [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) plugin for code, just for spacing issues.

Author: i thought you meant the pre-commit module...
i have it now

```bash
ros2 launch clr_mujoco_config clr_mujoco.launch.py
```

> 2. `pick_and_place_behavior` launch file setups up the MoveIt's `move_group`, color blob detection node, and behavior tree executor.

Reviewer: ```suggestion
2. Launches nodes for MoveIt's `move_group`, color blob detection, and behavior tree executor.
```

```bash
ros2 launch clr_behavior_pick_and_place_demo pick_and_place_behavior.launch.py
```

### Running the demo
To successfully pick up the bag you have to position the wrist camera above the cargo transfer bag where it can see the red handle.

> You can do that by calling

Reviewer: consistency

```suggestion
You can do that by calling:
```

```bash
ros2 action send_goal /bt_execution btcpp_ros2_interfaces/action/ExecuteTree "{target_tree: 'RandomStartTransform'}"
```
When ready you can execute the `BagPickUp` behavior tree.
```bash
ros2 action send_goal /bt_execution btcpp_ros2_interfaces/action/ExecuteTree "{target_tree: 'BagPickUp'}"
```

### Additional Notes
The setup process spins up a behavior tree executor node responsible for running the demo properly.

The behavior tree executor provides ROS service to list all of the available trees, and a ROS action to execute the registered trees.

> To get the list of all available tree:

Reviewer: ```suggestion
To get the list of all available trees:
```
